pkg/secrets-store/provider_client.go

```diff
@@ -154,6 +155,36 @@ func (p *PluginClientBuilder) Cleanup() {
 	p.conns = make(map[string]*grpc.ClientConn)
 }
 
+// EnableHealthCheck enables periodic healthcheck for configured provider clients by making
+// a Version() RPC call. If the provider healthcheck fails, we log an error.
+func (p *PluginClientBuilder) EnableHealthCheck(interval time.Duration, stopCh <-chan struct{}) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-ticker.C:
+			p.lock.RLock()
+
+			for provider, client := range p.clients {
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+
+				runtimeVersion, err := Version(ctx, client)
```

Comment:
No need to change in this PR, but the service config defined above could be modified to add default request timeouts to the entire service or per-method:

https://github.com/grpc/grpc/blob/master/doc/service_config.md